### PR TITLE
Added the ability to copy imported assets automatically to the Resources dir

### PR DESCRIPTION
**Pull request by SeleDreams**
# PR Details

This pull request adds a prompt to copy imported assets to the Resources directory directly from the editor.

## Description
- I added a prompt asking if the user wishes to copy the file they imported to the Resources directory.
- If they say yes, another prompt asks them if they want to copy it to the default directory, if they say yes, it will copy it to the default Resources directory following the structure of the Assets dir (if they import the file TheSong.mp3 in Assets/Music the resource would get copied to Resources/Music/TheSong.mp3), else, a save file dialog will appear allowing them to choose where in the resources directory they want to place the file.
- If they say no, the file will get imported as is.
- If the resource already exists, a prompt will appear allowing them to either override the file, either cancel the operation.

## Related Issue
#1633 

## Motivation and Context

This change will avoid new users breaking their projects by incorrectly importing their assets. This improvement makes the workflow more intuitive.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x ] My change requires a change to the documentation. 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by Aggror** [comment]
Is this related to this one?
https://github.com/stride3d/stride/pull/1781

**Comment by SeleDreams** [comment]
> Is this related to this one? #1781

It's also related to this one as it builds upon it yes

**Comment by Aggror** [comment]
Nice work!

File: sources/editor/Stride.Core.Assets.Editor/ViewModel/AssetCollectionViewModel.cs [hunk 1]
```diff
@@ -537,31 +535,99 @@ private string ComputeNamespace(DirectoryBaseViewModel directory)
...
-        private async Task<List<AssetViewModel>> InvokeAddAssetTemplate(LoggerResult logger, string name, DirectoryBaseViewModel directory, TemplateAssetDescription templateDescription, IEnumerable<UFile> files, PropertyContainer? customParameters)
+        private async Task<string> GetAssetCopyDirectory(DirectoryBaseViewModel directory, UFile file)
         {
-            List<AssetViewModel> newAssets = new List<AssetViewModel>();
-            if (files is not null)
-            {
-                foreach (var file in files)
+            string path = directory.Path;
+            var message = Tr._p("Message", "Do you want to place the resource in the default location ?").ToFormat(file.FullPath);
+            string finalPath = directory.Package.Package.ResourceFolders[0] + "/" + path + file.GetFileName();
```
Comment (Eideren) [inline sources/editor/Stride.Core.Assets.Editor/ViewModel/AssetCollectionViewModel.cs:542]: Could use the `Path` API, `Path.Combine` for example

File: sources/editor/Stride.Core.Assets.Editor/ViewModel/AssetCollectionViewModel.cs [hunk 2]
```diff
@@ -537,31 +535,99 @@ private string ComputeNamespace(DirectoryBaseViewModel directory)
...
+            if (pathResult == MessageBoxResult.No)
+            {
+                bool done = false;
+                do
                 {
-                    bool inResourceFolder = false;
-                    foreach (var resourceFolder in directory.Package.Package.ResourceFolders)
+                    var fileDialog = Dialogs.CreateFileSaveModalDialog();
+                    fileDialog.Filters = new List<FileDialogFilter>() { new FileDialogFilter("", file.GetFileExtension()) };
+                    fileDialog.InitialDirectory = directory.Package.Package.ResourceFolders.FirstOrDefault().FullPath.Replace("/", "\\");
```
Comment (Eideren) [inline sources/editor/Stride.Core.Assets.Editor/ViewModel/AssetCollectionViewModel.cs:551]: While the editor doesn't run on anything else but windows for now, might as well reduce issues we might have when migrating.
https://learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/8.0/file-path-backslash
> Backslash ( \ ) characters are valid in directory and file names on Unix

You can use `Path.GetFullPath(myPath)` to convert into the OS's directory separator.

File: sources/editor/Stride.Core.Assets.Editor/ViewModel/AssetCollectionViewModel.cs [hunk 3]
```diff
@@ -537,31 +535,99 @@ private string ComputeNamespace(DirectoryBaseViewModel directory)
...
+
+                    // If the user closes the dialog, assume that they want to use the default directory
+                    if (result != DialogResult.Ok)
+                    {
+                        done = true;
+                    }
+                    else
                     {
-                        if (file.FullPath.StartsWith(resourceFolder.FullPath))
+                        var fullPath = Path.GetFullPath(fileDialog.FilePath).Replace("\\", "/");
```
Comment (Eideren) [inline sources/editor/Stride.Core.Assets.Editor/ViewModel/AssetCollectionViewModel.cs:562]: Should skip the replace under non-windows, you can test against `Path.DirectorySeparatorChar` or os directly. The output path should already be using forward slash in those OS for dir separator.
Without that check, if the directory name or file name contains a backslash in those platforms it'll break.

File: sources/editor/Stride.Core.Assets.Editor/ViewModel/AssetCollectionViewModel.cs [hunk 4]
```diff
@@ -537,31 +535,99 @@ private string ComputeNamespace(DirectoryBaseViewModel directory)
...
+                                    
+                                    // Abort if the user says no or closes the prompt
+                                    if (copyResult != MessageBoxResult.Yes)
+                                    {
+                                        return newAssets;
+                                    }
+                                    File.Delete(finalPath);
+                                }                                
+                                
+                                File.Copy(file.FullPath, finalPath);
```
Comment (Eideren) [inline sources/editor/Stride.Core.Assets.Editor/ViewModel/AssetCollectionViewModel.cs:619]: Maybe copy with overwrite [Copy(String, String, Boolean)](https://learn.microsoft.com/en-us/dotnet/api/system.io.file.copy?view=net-7.0#system-io-file-copy(system-string-system-string-system-boolean)) when it exists, and use normal copy if the file doesn't.
That way it's an atomic operation, if a file somehow moves there between the delete and copy it won't fail. 
Very minor though, feel free to ignore.

**Review by Eideren — CHANGES_REQUESTED** [review]
Thanks a bunch @SeleDreams ! There's a couple of minor things I would like to go over, but otherwise it looks great !

**Comment by SeleDreams** [comment]
I changed some stuff based on the suggestions from @Eideren

**Comment by Eideren** [comment]
Thanks a bunch @SeleDreams !

**Pull request merged**